src/Evolution/Systems/ForceFree/Subcell/NeighborPackagedData.hpp

```diff
@@ -0,0 +1,288 @@
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "Domain/Structure/Side.hpp"
+#include "Domain/Tags.hpp"
+#include "Domain/TagsTimeDependent.hpp"
+#include "Evolution/BoundaryCorrectionTags.hpp"
+#include "Evolution/DgSubcell/Projection.hpp"
+#include "Evolution/DgSubcell/Reconstruction.hpp"
+#include "Evolution/DgSubcell/Tags/GhostDataForReconstruction.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/DgSubcell/Tags/OnSubcellFaces.hpp"
+#include "Evolution/DgSubcell/Tags/SubcellOptions.hpp"
+#include "Evolution/DiscontinuousGalerkin/Actions/PackageDataImpl.hpp"
+#include "Evolution/DiscontinuousGalerkin/NormalVectorTags.hpp"
+#include "Evolution/Systems/ForceFree/FiniteDifference/Factory.hpp"
+#include "Evolution/Systems/ForceFree/FiniteDifference/ReconstructWork.hpp"
+#include "Evolution/Systems/ForceFree/FiniteDifference/Reconstructor.hpp"
+#include "Evolution/Systems/ForceFree/FiniteDifference/Tags.hpp"
+#include "Evolution/Systems/ForceFree/Fluxes.hpp"
+#include "Evolution/Systems/ForceFree/Subcell/ComputeFluxes.hpp"
+#include "Evolution/Systems/ForceFree/System.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/CallWithDynamicType.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ForceFree::subcell {
+
+/*!
+ * \brief On elements using DG, reconstructs the interface data from a
+ * neighboring element doing subcell.
+ *
+ * The neighbor's packaged data needed by the boundary correction is computed
+ * and returned so that it can be used for solving the Riemann problem on the
+ * interfaces.
+ *
+ * Note that for strict conservation the Riemann solve should be done on the
+ * subcells, with the correction being projected back to the DG interface.
+ * However, in practice such strict conservation doesn't seem to be necessary
+ * and can be explained by that we only need strict conservation at shocks, and
+ * if one element is doing DG, then we aren't at a shock.
+ */
+struct NeighborPackagedData {
+  template <typename DbTagsList>
+  static FixedHashMap<maximum_number_of_neighbors(3),
+                      std::pair<Direction<3>, ElementId<3>>, DataVector,
+                      boost::hash<std::pair<Direction<3>, ElementId<3>>>>
+  apply(const db::DataBox<DbTagsList>& box,
+        const std::vector<std::pair<Direction<3>, ElementId<3>>>&
+            mortars_to_reconstruct_to) {
+    using evolved_vars_tags = typename System::variables_tag::tags_list;
+    using fluxes_tags = typename Fluxes::return_tags;
+
+    // subcell currently does not support moving mesh
+    ASSERT(not db::get<domain::Tags::MeshVelocity<3>>(box).has_value(),
```

Comment:
There is moving mesh now for GrMhd subcell, so why this restriction?

Reply:
It's just that I haven't implemented that for ForceFree yet... 😓

Reply:
Shall I apply moving mesh and update this PR?

Reply:
for now you can just update the comment to say that subcell with moving mesh isn't implemented for this system yet

Reply:
Updated